### PR TITLE
GEN-1835 - feat(AverageRatingBanner): use real data

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Extract share pillow UI used in `ReviewsDialogV2` into a separated component: `PillowHeader`
* Refactor: rename `features/productReviews` --> `features/memberReviews` and add functions for getting _company_ (general) average rating and _latest reviews_ (involving multiple products)
* Update `AverageRatingBanner` so it uses real data

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/ocPE5OGCo5Y6fhJekeDQ/9b20d45b-0370-4d20-a3b8-f15a0e572ca8.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/ocPE5OGCo5Y6fhJekeDQ/9b20d45b-0370-4d20-a3b8-f15a0e572ca8.mov">Screen Recording 2024-02-22 at 10.24.36.mov</video>

## Justify why they are needed

Part of [_Product Reviews v2_](https://www.figma.com/file/PUA7SuN75sWH4Q1QxDaeCJ/Product-Reviews?type=design&node-id=1447-1775&mode=design&t=Sri79oZDuQ3Tbqo4-4)

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* **#3960** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3960?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3960?utm_source=stack-comment).

**Comment by guilhermespopolin** [comment]
**Discussion Point**

In order for this to work without having to deal with empty/loading state (while fetching the data client side) we need to fetch _latest reviews_ and _average rating_ for every page. That ofc increases the amount of data we need to send to the client for hydration. This new addition raises some warnings from next:

Screenshot 2024-02-22 at 10 39 05
Screenshot 2024-02-22 at 10 37 00

I remember seen those even before this addition but it makes things worse as it adds more stuff. That happens because, for product pages we're fetching/adding product specific _latest reviews_ and _average rating_ info + general _latest reviews_ and _average rating_ which is currently used by `AverageRatingBannerBlock`.

**Some brain dump**

1. We don't expected to render `AverageRatingBannerBlock` for _Product Pages_ so if we don't fetch general _latest reviews_ and _average rating_ for them that payload size would be reduced. The con however is that we're kind of breaking the flexibility of blocks since in theory, they are supposed to render on whatever page editors want.
2. If we fetch that data client side for `AverageRatingBannerBlock` the payload size would also be reduced. However component code would be more complex since we'd need to include client side data fetching + loading/empty state management.
3. Since we're moving into _app router_ we probably could keep it that way since that warn would naturally goes away as we start to make usage of server components. But until there, we might be hurting performance for the most important pages of our app: _Product Pages_.

Do you guys have any thoughts about it?
@gustaveen @alebedev

File: apps/store/src/features/memberReviews/memberReviews.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+// Data format and used keys are defined in the cloud function:
+// https://shorturl.at/eoQRV
+const KV_AVERAGE_RATING_KEY = 'averageRating'
+const KV_LATEST_REVIEWS_KEY = 'latestReviews'
+const KV_AVERAGE_RATING_BY_PRODUCT_KEY = 'averageRatingByProduct'
+const KV_LATEST_REVIEWS_BY_PRODUCT_KEY = 'latestReviewsByProduct'
+
+export const fetchReviewsData = async (productId?: string): Promise<ReviewsData | null> => {
+  try {
+    const reviewsData = await (productId
```
Comment (alebedev) [inline apps/store/src/features/memberReviews/memberReviews.ts:23]: Is this abstraction worth it?  We can use `fetchProductReviewsData` / `fetchCompanyReviewsData` directly in a more type safe way

File: apps/store/src/features/memberReviews/memberReviews.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      return null
+    }
+
+    return {
+      averageRating,
+      reviewsByScore,
+      reviewsDistribution: generateReviewsDistribution(reviewsByScore),
+    }
+  } catch (error) {
+    throw new Error(`Failed to fetch product reviews data for product ${productId}`)
```
Comment (alebedev) [inline apps/store/src/features/memberReviews/memberReviews.ts:142]: Original error suppressed here, do we want to log it?

File: apps/store/src/features/memberReviews/memberReviews.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  reviewCommentsSchema,
+  commentByScoreSchema,
+  validateLatestReviews,
+  validateProductReviewsComments,
+  validateAverageRating,
+} from './memberReviews.utils'
+
+// Data format and used keys are defined in the cloud function:
+// https://shorturl.at/eoQRV
+const KV_AVERAGE_RATING_KEY = 'averageRating'
```
Comment (alebedev) [inline apps/store/src/features/memberReviews/memberReviews.ts:16]: What do you think about grouping keys? Something like `KV_KEY.AVERAGE_RATING` is probably easier to read

File: apps/store/src/features/memberReviews/memberReviews.ts [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+  const validationResult = validateProductReviewsComments(latestProductReviews)
+  if (!validationResult.success) {
+    console.log(`Member Reviews |  validation error: ${validationResult.error}`)
+    return null
+  }
+
+  return parseProductReviews(latestProductReviews)
+}
+
+const parseProductReviews = (
```
Comment (alebedev) [inline apps/store/src/features/memberReviews/memberReviews.ts:212]: Same as above:
- parse
- reduce

**Review by alebedev — APPROVED** [review]
Good job!  Some code styling comments, nothing major

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Feb 22, 9:39 AM EST**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3960).

**Pull request merged**